src/query/sql/src/planner/optimizer/hyper_dp/dphyp.rs

```diff
@@ -178,6 +178,12 @@ impl DPhpy {
                 }
                 // Add join conditions
                 for condition_pair in op.left_conditions.iter().zip(op.right_conditions.iter()) {
+                    let left_used_tables = condition_pair.0.used_tables()?;
+                    let right_used_tables = condition_pair.1.used_tables()?;
+                    if left_used_tables.is_empty() || right_used_tables.is_empty() {
```

Comment:
When will this happen ?

Reply:
For example, the join key is from the derived column generated during decorrelation. 

```
explain select t.number from numbers(2) as t where number = (select * from numbers(1) where number = 0)
----
HashJoin
├── output columns: [t.number (#0)]
├── join type: INNER
├── build keys: [_if_scalar_subquery (#4)]
├── probe keys: [CAST(t.number (#0) AS UInt64 NULL)]
```